go/samples/prompts/main.go

```diff
@@ -328,6 +329,47 @@ func PromptWithMediaType(ctx context.Context, g *genkit.Genkit) {
 	fmt.Println(resp.Text())
 }
 
+/*
+Inline schema equivalence
+
+	  title: string, recipe title
+	  ingredients(array):
+		name: string
+		quantity: string
+	  steps(array, the steps required to complete the recipe): string
+*/
+type Ingredient struct {
+	Name     string `json:"name" description:"ingredient name"`
+	Quantity string `json:"quantity" description:"ingredient quantity"`
+}
+
+type RecipeSchema struct {
+	Title       string       `json:"title" description:"Recipe name"`
+	Ingredients []Ingredient `json:"ingredients" description:"Recipe ingredients"`
+	Steps       []string     `json:"steps" description:"Recipe steps"`
+}
+
+func PromptWithSchema(ctx context.Context, g *genkit.Genkit) {
+	err := genkit.DefineSchema(g, "recipe", RecipeSchema{})
+	if err != nil {
+		log.Fatalf("error defining schema: %v", err)
+	}
```

Comment:
```suggestion
	genkit.DefineSchema(g, "recipe", RecipeSchema{})
```

This should panic instead of returning an error.